frontend/components/puzzle4.jsx

```diff
@@ -2,8 +2,38 @@ import React from 'react';
 
 import PuzzleHeader from './puzzleHeader';
 
+import {
+  KeypadContainer, Button, ButtonContainer } from './buttonContants';
+
 class Puzzle4 extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: [false, false, false, false, false, false, false, false, false]
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+    this.clearValue = this.clearValue.bind(this);
+  }
+
+  handleClick(event) {
+    let buttonNum = parseInt(event.target.value) - 1;
```

Comment:
instead of parsing the value, can we pass in the value and return a new function that uses it?
```
handleClick(value) {
  return () => {
    this.props.send(... { answer: value });
  }
}
```